pos_stock_picking_invoice_link/models/pos_order.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    def _prepare_invoice(self):
+        res = super(PosOrder, self)._prepare_invoice()
+        res.update({
+            'picking_ids': [(6, 0, self.picking_id.ids)],
+        })
+        return res
+
+    def _action_create_invoice_line(self, line=False, invoice_id=False):
+        invoice_line = super(
+            PosOrder, self)._action_create_invoice_line(line, invoice_id)
+        if not line:
+            return invoice_line
+        move = self.env['stock.move'].search([
+            ('picking_id', '=', self.picking_id.id),
+            ('name', '=', line.name)])
```

Comment:
Please use a harder link, like `product_id`. Any way, we will have the problem when the same product is put several times (for example, when using different lots). Isn't any deterministic way of linking a pos line with an stock move?

Reply:
Well, only a move is created by each line: https://github.com/odoo/odoo/blob/10.0/addons/point_of_sale/models/pos_order.py#L690

Isn't it deterministic enough?

Reply:
I'm afraid not. As said, this will find multiple moves if you add several times a product, linking all of them to all lines.

POS line 1: Product 1 > Move line 1
POS line 2: Product 2 > Move line 2
POS line 3: Product 1 > Move line 3

Resulting in:

Invoice line 1 from POS line 1, linked to Move line 1 and 3
Invoice line 2 from POS line 2, linked to Move line 2
Invoice line 3 from POS line 3, linked to Move line 1 and 3

Reply:
I made a test with your example to show you that doesn't happen. Every pos order line has a unique sequence (determined by an ir.sequence) and its consequent stock move is named after it. So it's very unlikely that we're getting to moves with that premise, isn't it?

Reply:
I don't see in that commit the test you are mentioning, but I'm interpreting from your words that picking text is composed by not the product name, but another thing?